R/insert-ui.R

```diff
@@ -172,3 +172,275 @@ removeUI <- function(selector,
   if (!immediate) session$onFlushed(callback, once = TRUE)
   else callback()
 }
+
+
+#' Dynamically add a tabPanel
+#'
+#' Dynamically add a \code{tabPanel()} into an existing \code{tabsetPanel}.
+#'
+#' @param tabsetPanelId The \code{id} of the \code{tabsetPanel()} into which
+#'   \code{tab} will be inserted.
+#'
+#' @param tab The tab element to be added (must be created with
+#'   \code{tabPanel}).
+#'
+#' @param target The \code{value} of an existing \code{tabPanel()}, next to
+#'   which \code{tab} will be added. If \code{NULL},the \code{tab} will be
+#'   placed either as the first tab or the last tab, depending on the
+#'   \code{position} argument.
+#'
+#' @param position Should \code{tab} be added to the right ot to the left
+#'   of \code{target}?
+#'
+#' @param immediate Whether \code{tab} should be immediately inserted into
+#'   the app when you call \code{insertTab}, or whether Shiny should wait until
+#'   all outputs have been updated and all observers have been run (default).
```

Comment:
Yeah, it definitely came from there. This is actually something I wanted to ask you Joe, i.e. if it's necessary to keep this arg for the tab functions. From your reply above, I'm guessing "no"? In which case, I just always call `session$onFlushed(callback, once = TRUE)` instead of giving the user an option... I think this is what makes sense, but I just wanted to confirm